pulsar-function-go/conf/conf.go

```diff
@@ -87,16 +87,19 @@ func (c *Conf) GetConf() *Conf {
 		flag.Usage()
 	}
 
+	if confContent == "" && confFilePath == "" {
+		log.Errorf("no yaml file or conf content provided")
+		return nil
+	}
+
 	if confFilePath != "" {
 		yamlFile, err := ioutil.ReadFile(confFilePath)
-		if err != nil {
-			log.Errorf("not found conf file, err:%s", err.Error())
-			return nil
-		}
-		err = yaml.Unmarshal(yamlFile, c)
-		if err != nil {
-			log.Errorf("unmarshal yaml file error:%s", err.Error())
-			return nil
+		if err == nil {
```

Comment:
there are multiple reasons that ReadFile can fail. it is not just FileNotExist. you will need to handle err and check if the err is FileNotExist or not.

Reply:
how about 
```
		if err == nil {
			err = yaml.Unmarshal(yamlFile, c)
			if err != nil {
				log.Errorf("unmarshal yaml file error:%s", err.Error())
				return nil
			}
		} else if (err != nil || os.IsNotExist(err)) && confContent == "" {
			log.Errorf("not found conf file, err:%s", err.Error())
			return nil
		}
```
so when `ioutil.ReadFile` returns err and `confContent` empty, panic caused, otherwise go load `confContent`.

Reply:
sgtm